pyro/infer/mcmc/hmc.py

```diff
@@ -233,29 +349,18 @@ def setup(self, *args, **kwargs):
             if node["fn"].support is not constraints.real and self._automatic_transform_enabled:
                 self.transforms[name] = biject_to(node["fn"].support).inv
                 site_value = self.transforms[name](node["value"])
-            r_loc = site_value.new_zeros(site_value.shape)
-            r_scale = site_value.new_ones(site_value.shape)
-            self._r_dist[name] = dist.Normal(loc=r_loc, scale=r_scale)
-        self._validate_trace(trace)
-
-        if self.adapt_step_size:
-            self._adapt_phase = True
-            z = {name: node["value"] for name, node in self._iter_latent_nodes(trace)}
-            for name, transform in self.transforms.items():
-                z[name] = transform(z[name])
-            with pyro.validation_enabled(False):
-                self.step_size = self._find_reasonable_step_size(z)
-            self.num_steps = max(1, int(self.trajectory_length / self.step_size))
-            # make prox-center for Dual Averaging scheme
-            loc = math.log(10 * self.step_size)
-            self._adapted_scheme = DualAveraging(prox_center=loc)
+            self._r_shapes[name] = site_value.shape
+            self._r_numels[name] = site_value.numel()
 
-    def end_warmup(self):
-        if self.adapt_step_size:
-            self._adapt_phase = False
-            _, log_step_size_avg = self._adapted_scheme.get_state()
-            self.step_size = math.exp(log_step_size_avg)
-            self.num_steps = max(1, int(self.trajectory_length / self.step_size))
+        self._validate_trace(trace)
+        mass_matrix_size = sum(self._r_numels.values())
+        if self.full_mass:
+            self._inverse_mass_matrix = eye_like(site_value, mass_matrix_size)
+        else:
+            self._inverse_mass_matrix = site_value.new_ones(mass_matrix_size)
+        self._update_r_dist()
```

Comment:
Since we are assuming an identity mass matrix by default, would it simplify the code if we used that as the argument (`mass_matrix_structure="identity"`), instead of maintaining two boolean args `adapt_mass_matrix` and `full_max`?

Reply:
I think that maintaining `adapt_mass_matrix` has longer benefit. Later, we can let users set initial `inverse_mass_matrix` and decide if they want to adapt it or not.